src/database/schema/ChannelListSchema.ts

```diff
@@ -239,14 +285,56 @@ class ChannelList implements BaseDbSchema {
     return 'channel_lists';
   }
 
-  static fromWebsocketObject(json): ChannelList {
+  static updateChannelDescription = async (
+    db: SQLiteDatabase,
+    channelId: string,
+    description: string,
+    json,
+    isUpdateTimestamp = false
+  ) => {
+    let rawJson: string | null = null;
+
+    try {
+      rawJson = JSON.stringify(json);
+    } catch (e) {
+      console.log('error stringify:', e);
+    }
+
+    try {
+      const queryTimestamp = `UPDATE ${ChannelList.getTableName()}
+        SET description = ?, created_at = ?, last_updated_at = ?, raw_json = ?
+        WHERE id = ?;`;
+      const queryWithoutTimestamp = `UPDATE ${ChannelList.getTableName()}
+        SET description = ?, raw_json = ?
+        WHERE id = ?;`;
+
+      const replacementTimestamp = [
+        description,
+        new Date().toISOString(),
+        new Date().toISOString(),
+        rawJson,
+        channelId
+      ];
+      const replacementWithoutTimestamp = [description, rawJson, channelId];
+
+      if (isUpdateTimestamp) {
+        await db.executeSql(queryTimestamp, replacementTimestamp);
+      } else {
+        await db.executeSql(queryWithoutTimestamp, replacementWithoutTimestamp);
+      }
+    } catch (e) {
+      console.log('error updating channel description', e);
+    }
+  };
+
+  static fromWebsocketObject(json, channelType: ChannelType): ChannelList {
     return new ChannelList({
       id: json?.channel?.id,
       channelPicture: json?.targetImage,
       name: json?.targetName,
       description: json?.message?.text || json?.message?.message,
       unreadCount: json?.unread_count,
-      channelType: 'ANON_PM',
+      channelType,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `fromWebsocketObject` function now accepts a `channelType` parameter which is a good improvement for flexibility and reusability of the function. The function implementation looks correct.



<!-- This is an auto-generated comment by OSS CodeRabbit -->